cmd/controller-init/main.go

```diff
@@ -57,15 +63,62 @@ func main() {
 		logger.Fatalw("Failed to get cluster config", zap.Error(err))
 	}
 
+	// Check into the config-secrets. Its *possible* that the ConfigMap
+	// doesn't exist. This is because the ConfigMap has to point at a secret
+	// that an operator HAS to create. Therefore, we need to create a
+	// placeholder ConfigMap if one doesn't exist.
+	corev1Client, err := clientcorev1.NewForConfig(clusterConfig)
+	if err != nil {
+		logger.Fatalw("Failed to get the corev1 client set", zap.Error(err))
+	}
+	secretsCM, err := corev1Client.
+		ConfigMaps(v1alpha1.KfNamespace).
+		Get(sourceconfig.SecretsConfigName, metav1.GetOptions{})
+
+	switch {
+	case apierrs.IsNotFound(err):
+		// ConfigMap isn't found, so we should create one.
+		logger.Warnw("ConfigMap %q not found. Creating a placeholder ConfigMap...", sourceconfig.SecretsConfigName)
+		if err := createPlaceholderConfigMap(corev1Client); err != nil {
+			logger.Fatalw("Failed to create ConfigMap %q: %v", sourceconfig.SecretsConfigName, err)
+		}
+	case err != nil:
+		logger.Fatalw("Failed to fetch ConfigMap %q: %v", sourceconfig.SecretsConfigName, err)
+	case secretsCM.Data[sourceconfig.BuildImagePushSecretKey] != "":
```

Comment:
I don't think it should be the responsibility of the init controller to check this since it can dynamically change later on. Instead, a sensible default should be allowed e.g. blank = no secrets.

If we _must_ check here, it should be done using the same logic as the constructor of the config object to ensure as we add additional values/change logic this bit doesn't become outdated